File: engine/model/src/main/scala/tools/packet/field/effect/TrembleEffect.scala
package tools.packet.field.effect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class TrembleEffect(private var _type: Int, private var _delay: Int) extends PacketInput {
  def theType: Int = _type

  def delay: Int = _delay

  override def opcode(): SendOpcode = SendOpcode.FIELD_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/field/effect/TrembleEffect.java
package tools.packet.field.effect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record TrembleEffect(Integer theType, Integer delay) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.FIELD_EFFECT;
   }
}